data/sea/54-zebra.h

```diff
@@ -79,6 +103,7 @@ static ierror_msg_t zebra_translate
     /* temporary fix until icicle uses the same time as zebra */
     itime_t zebra_chord_time = itime_to_epoch_seconds (chord_time);
 
+    /* the number of facts before chord time for this entity */
     for (iint_t i = 0; i != row_count; ++i) {
```

Comment:
this is counting all the facts for every call, but could you just loop from `state->entity_fact_offset` up to `state->entity_fact_offset + zebra_chunk_fact_count`?